targets/TARGET_ONSEMI/TARGET_NCS36510/gpio_irq_api.c

```diff
@@ -193,42 +185,35 @@ void gpio_irq_free(gpio_irq_t *obj)
  */
 void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable)
 {
-
-    /* Enable the GPIO clock */
+    /* Enable the GPIO clock which may have been switched off by other drivers */
     CLOCK_ENABLE(CLOCK_GPIO);
-
+    obj->GPIOMEMBASE->IRQ_EDGE = obj->pinMask;
+    
     switch(event) {
-        case IRQ_RISE:
-            obj->GPIOMEMBASE->IRQ_EDGE = (obj->pinMask);
-            obj->GPIOMEMBASE->IRQ_LEVEL = (IO_ALL ^ (obj->pinMask));
-            /* Enable is an integer; hence checking for 1 or 0*/
-            if (enable == 1) {
-                /* Enable rising edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_SET = (obj->pinMask);
-            } else if (enable == 0) {
-                /* Disable rising edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_SET = (IO_ALL ^ (obj->pinMask));
-            }
-            break;
+        case IRQ_RISE: 
+          
+            /* Enable rising edge */
+            obj->GPIOMEMBASE->IRQ_POLARITY_SET = obj->pinMask;
+           break;
 
         case IRQ_FALL:
-            obj->GPIOMEMBASE->IRQ_EDGE = (obj->pinMask);
-            obj->GPIOMEMBASE->IRQ_LEVEL = (IO_ALL ^ (obj->pinMask));
-            /* Enable is an integer; hence checking for 1 or 0*/
-            if (enable == 1) {
-                /* Enable falling edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_CLEAR = (obj->pinMask);
-            } else if (enable == 0) {
-                /* Disable falling edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_CLEAR = (IO_ALL ^ (obj->pinMask));
-            }
+
+            /* Enable falling edge */
+            obj->GPIOMEMBASE->IRQ_POLARITY_CLEAR = obj->pinMask;
             break;
 
         default:
             /* No event is set */
             break;
     }
+    /* Enable the IRQ based on enable parameter */
+    if (enable == 1) {
```

Comment:
I personally would simplify further to just `if (enable) {} else {}`, on the assumption that it is supposed to be a boolean, even if not declared as that. But this is fine.